### PR TITLE
Refactor & tidy

### LINKED ISSUE #101: genus and higher level taxon rank detecting
```
create_taxonomic_update_lookup( 
+   taxa = c(
+     "Banksia integrifolia",
+     "Acacia longifolia",
+     "Commersonia rosea",
+     "Acacia sp."
+     ),
+   resources = resources
+ )
Checking alignments of 4 taxa

  -> 0 names already matched; 0 names checked but without a match; 4 taxa yet to be checked
# A tibble: 4 × 7
  original_name        aligned_name         accepted_name taxon_rank author aligned_reason updated_reason
  <chr>                <chr>                <chr>         <chr>      <chr>  <chr>          <chr>         
1 Banksia integrifolia Banksia integrifolia Banksia inte… Species    L.f.   match_06. Aut… accepted      
2 Acacia longifolia    Acacia longifolia    Acacia longi… Species    (Andr… match_06. Aut… accepted      
3 Commersonia rosea    Commersonia rosea    Androcalva r… Species    (S.A.… match_06. Aut… basionym      
4 Acacia sp.           Acacia sp. [XXXX]    NA            NA         NA     match_01. Rew… known_name_bu…
```

### LINKED ISSUE #106: make input names consistent across main functions
The first argument for `align_taxa` is called `original_name`, while for `create_taxonomic_update_lookup` it's `taxa`. 

Argument name need to be consistent across the main functions: `align_taxa` is called , `create_taxonomic_update_lookup` and `update_taxonomy`, also with outputs

For this argument I suggest `original_name` is better.

### LINKED ISSUE #128: Variable names ending with `_clean` should be `_aligned`
`update_taxonomy` creates several variables that end with `_clean`. Yet all of these variables are actually based on the `aligned_name` not the `cleaned_name`. The cleaned name is simply the original name run through `standardise_names`, before any name matches are made. I think it would be better to replace all the `_clean` suffixes with `_aligned`.

### LINKED ISSUE #129: error when zero species in a call match
Screenshot 2023-08-22 at 2 26 05 pm

### LINKED ISSUE #135: Tests reorganisation and restructruring
Tests usually organised by the main function it is testing, e.g. test-create_tax_update_looking.R, test-update_taxonomy. Need to organise tests within files with a section of "normal" use cases and then edge cases (name completely changes, subspecies, mulitple cases where taxonomists would be interested)

- Acceptance criteria/ What is considered "done" here: test directory is easy to navigate, organised, structured from 'simple cases' to more complex

Image

### PR DESCRIPTION
**Pull request by dfalster**
Reorganise files, functions and tests

- seperate update_taxonomy into component functions
- move code for taxon splitting into update_taxonomy
- expand testing (#118), reorganise tests to be more intuitive
- move some functions into their own seperate files
- remove temporary variables
- simplify update_taxonomy where possible
- renamed variables:
   - `taxonomic_reference` -> `taxonomic_dataset`
   - variables ending `_clean` -> `_align` (#128)

Fixes #129,  Fixes #101, Fixes #128 Fixes #106 Fixes #135 

Contributes to #120 #127 #110 #118

### CODE REVIEW

File: R/standardise_names.R [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    f("\\ssp(\\s|$)",   " sp. ") %>%
+    f("\\sspp.(\\s|$)", " sp. ") %>%
+    f("\\sspp(\\s|$)",  " sp. ") %>%
+    
+    ## subsp. not ssp, ssp., subsp or sub sp.
+    f("\\sssp(\\s|$)",     " subsp. ") %>%
+    f("\\sssp.(\\s|$)",    " subsp. ") %>%
+    f("\\ssubsp(\\s|$)",   " subsp. ") %>%
+    f("\\ssub sp.(\\s|$)", " subsp. ") %>%
+    f("\\sssp.(\\s|$)",    " subsp. ") %>%
```
Comment (ehwenk) [inline R/standardise_names.R:54]: the next two lines are duplicates from above - I'd already caught this, but had forgotten to make the change. I'll do it on the next branch

File: R/update_taxonomy.R [hunk 2]
```diff
@@ -58,70 +56,63 @@
...
-      mutate(
-        taxonomic_dataset_tmp = stringr::word(taxonomic_reference, 1),
-        taxonomic_rank_tmp = ifelse(species_and_infraspecific(taxon_rank), "species_and_infraspecific_taxa", taxon_rank),
-        row_number = dplyr::row_number()
-      ) %>%
-    split(paste(.$taxonomic_dataset_tmp, .$taxonomic_rank_tmp))
+    split(aligned_data,
+      paste(
+        # Taxonomic reference
+        stringr::word(aligned_data$taxonomic_dataset, 1),
```
Comment (ehwenk) [inline R/update_taxonomy.R:84]: I think line 84 is no longer required, since now the only values for taxonomic_dataset are "APC" and "APNI"

File: R/standardise_names.R [hunk 3]
```diff
@@ -0,0 +1,91 @@
+
+#' Standardise Taxon Names
+#'
+#' This function standardises taxon names by performing a series of text 
+#' substitutions to remove common inconsistencies in taxonomic nomenclature.
+#' The function takes a character vector of taxon names as input and returns a
+#' character vector of standardised taxon names as output.
```
Comment (wcornwell) [inline R/standardise_names.R:7]: standardised to what? I assume we're using APC as the gold standard for things like phrase names?

File: NAMESPACE [hunk 4]
```diff
@@ -5,7 +5,9 @@ export(create_species_state_origin_matrix)
 export(create_taxonomic_update_lookup)
 export(load_taxonomic_resources)
 export(native_anywhere_in_australia)
+export(standardise_names)
```
Comment (wcornwell) [inline NAMESPACE:8]: common R practice is to have synonyms with American spelling, but not sure how to do that

**Review by ehwenk — DISMISSED** [review]
Looks good.

File: R/standardise_names.R (hunk 3 above)
Comment (ehwenk) [reply R/standardise_names.R:7]: Yes, but what this does is ensure that all the different ways people might word "var.", "subsp." etc are standardised (to APC). To allow as many exact matches as possible before fuzzy matching. This speeds up processing a lot. @wcornwell wcornwell

File: NAMESPACE (hunk 4 above)
Comment (dfalster) [reply NAMESPACE:8]: Hmm, yes. But since it has Aus in the name,  I'm happy to leave it as Aus spelling

File: R/standardise_names.R (hunk 1 above)
Comment (dfalster) [reply R/standardise_names.R:54]: Please push before we merge

File: R/update_taxonomy.R (hunk 2 above)
Comment (dfalster) [reply R/update_taxonomy.R:84]: Please push before we merge

File: R/standardise_names.R (hunk 3 above)
Comment (dfalster) [reply R/standardise_names.R:7]: Lizzy, can you add some more text to describe how we a "standardising".

**Review by ehwenk — APPROVED** [review]
Looks good.

Add to your commit message that this also now solves issue #132  (default option for align_taxa to output a subset of columns); issue #133 (standardise taxon ranks)
